spec.md

```diff
@@ -547,212 +547,129 @@ See [@sec:mem] for more details.
-; ...
-```
-
-Inference rules are as follows:
+### Reset Types
 
-1. An uninferred reset driven by and/or driving only asynchronous resets will be inferred as asynchronous reset.
-1. An uninferred reset driven by and/or driving both asynchronous and synchronous resets is an error.
-1. Otherwise, the reset is inferred as synchronous (i.e. the uninferred reset is only invalidated or is driven by or drives only synchronous resets).
+Once a circuit is powered on, it may require an explicit reset in order to put it into a known state.
+For this, we use a reset type.
+In FIRRTL, we have the option of using both synchronous or asynchronous resets.
 
-`Reset`{.firrtl}s, whether synchronous or asynchronous, can be cast to other types.
-Casting between reset types is also legal:
+The synchronous reset type is simply a 1-bit unsigned integer: `UInt<1>`{.firrtl}.
+The asynchronous reset type is `AsyncReset`{.firrtl}.
 
-``` firrtl
-input a : UInt<1>
-output y : AsyncReset
-output z : Reset
-wire r : Reset
-connect r, a
-connect y, asAsyncReset(r)
-connect z, asUInt(y)
-```
-
-See [@sec:primitive-operations] for more details on casting.
+The reset types also have the uninferred form: `Reset`{.firrtl} (see [@sec:reset-inference]).
 
 ## Aggregate Types
 
-FIRRTL supports three aggregate types: vectors, bundles, and enumeration.
-Aggregate types are composed of ground types or other aggregate types.
+FIRRTL supports three aggregate types: vectors, bundles, and enumerations.
 
 ### Vector Types
 
 A vector type is used to express an ordered sequence of elements of a given type.
-The length of the sequence must be non-negative and known.
 
-The following example specifies a ten element vector of 16-bit unsigned integers.
+The following example specifies a 10-element vector of 16-bit unsigned integers.
 
 ``` firrtl
 UInt<16>[10]
 ```
 
-The next example specifies a ten element vector of unsigned integers of omitted but identical bit widths.
-
-``` firrtl
-UInt[10]
-```
+Note that the element type of a vector can be any type, including another aggregate types.
+The following example specifies a 20-element vector, each of which is a 10-element vector of 16-bit unsigned integers.
 
-Note that any type, including other aggregate types, may be used as the element type of the vector.
-The following example specifies a twenty element vector, each of which is a ten element vector of 16-bit unsigned integers.
+TODO: *Any* type? What about probes and properties?
 
-``` firrtl
+```firrtl
 UInt<16>[10][20]
 ```
 
+TODO: In analogy to 0-width integers, write a section on 0-length vecs.
+
 ### Bundle Types
 
-A bundle type is used to express a collection of nested and named types.
-All fields in a bundle type must have a given name, and type.
-All names must be legal identifiers.
+A bundle type is used to represent a collection of values.
+They can also be used to facilitate bidirectional connections between circuit components.
+
+A bundle type consists of a set of fields.
```

Comment:
Mega-nit: I grok this is an intuitional sentence. However, `set` makes me ask if this is a mathematical set and whether it is ordered (it is from the port-lowering ABI). Is there a better way of stating this without making it sound like a vector? Collection, sequence, e.g.?

Reply:
> A bundle type consists of multiple fields.

Changed.